File: Student Management System/student-management-frontend/src/services/StudentService.js
import axios from 'axios';

const BASE_URL = 'http://localhost:8080/api/students';

export const getAllStudents = () => axios.get(BASE_URL);
export const createStudent = (student) => axios.post(BASE_URL, student);
export const updateStudent = (id, student) => axios.put(`${BASE_URL}/${id}`, student);
export const deleteStudent = (id) => axios.delete(`${BASE_URL}/${id}`);

File: Student Management System/student-management/src/main/java/com/example/studentmanagement/service/StudentService.java
package com.example.studentmanagement.service;

import com.example.studentmanagement.model.Student;
import com.example.studentmanagement.repository.StudentRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class StudentService {

    @Autowired
    private StudentRepository studentRepository;

    // Create Student
    public Student createStudent(Student student) {
        return studentRepository.save(student);
    }

    // Get All Students
    public List<Student> getAllStudents() {
        return studentRepository.findAll();
    }

    // Get Student by ID
    public Student getStudentById(String id) {
        return studentRepository.findById(id).orElse(null);
    }

    // Update Student
    public Student updateStudent(String id, Student updatedStudent) {
        if (studentRepository.existsById(id)) {
            updatedStudent.setId(id);
            return studentRepository.save(updatedStudent);
        }
        return null;
    }

    // Delete Student
    public void deleteStudent(String id) {
        studentRepository.deleteById(id);
    }
}
